handler.go

```diff
@@ -55,9 +60,23 @@ func NewWASMServer(wasmFile string, args []string, coverageFile string, l *log.L
 		srv.envMap[vars[0]] = vars[1]
 	}
 
-	buf, err := os.ReadFile(path.Join(runtime.GOROOT(), "misc/wasm/wasm_exec.js"))
-	if err != nil {
-		return nil, err
+	var buf []byte
+	for _, loc := range wasmLocations {
+		buf, err = os.ReadFile(filepath.Join(runtime.GOROOT(), loc))
+		if err == nil {
+			break
+		}
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	}
+	if len(buf) == 0 {
+		if !strings.Contains(runtime.GOROOT(), "golang.org"+string(os.PathSeparator)+"toolchain") {
+			return nil, err
+		}
+		fmt.Fprintln(os.Stderr, "The go toolchain does not include the WebAssembly exec helper before Go 1.24.")
+		fmt.Fprintf(os.Stderr, "You should copy wasm_exec.js to %s\n", filepath.Join(runtime.GOROOT(), "misc", "wasm", "wasm_exec.js"))
+		os.Exit(1)
 	}
```

Comment:
What are your thoughts on simplifying this by:
- Directly matching the path instead of checking for GOROOT
- Directly returning the error and let the os.Exit happen at the upper layer

Something like:

```suggestion
	if err != nil {
		var perr *os.PathError
		if errors.As(err, &perr) {
			if strings.Contains(perr.Path, filepath.Join("golang.org", "toolchain")) {
				return nil, fmt.Errorf("The Go toolchain does not include the WebAssembly exec helper before Go 1.24. Please copy wasm_exec.js to %s", filepath.Join(runtime.GOROOT(), "misc", "wasm", "wasm_exec.js"))
			}
		}
		return nil, err
	}
```

Reply:
The Path included in PathError is basically the name passed into ReadFile, and the name is constructed with runtime.GOROOT already included, so I think it works, but I don't quite understand what difference it makes.